### DISCUSSION TITLE
Typing Issues with Pydantic and Ruff Linting

### DISCUSSION THREAD

**Discussion by ianhuezo** [encode/httpx#3256] [Ideas]
Is there a reasonable way to convert to a pydantic BaseModel using httpx?  I'm having a bit of trouble since ruff gives a type error of something like:
```python
response = httpx.get(...)
my_json = response.json()
model = MyPydanticModel(**my_json) #throws a typing error that `Any` can't be used
```

I used to be able to one-line it like in the FastAPI docs here:
```python
MyPydanticModel(**response.json())
```

Took a look at when that was changed in this PR #923, but it seemed like the user was making a few assumptions before conversion.  Namely, he was forgetting that the items may not exist yet where something simple like
```python
_ = data["items"] 
```
could evaluate to None.  Let me know if I'm missing something though.

**Comment by lovelydinosaur** [comment]
Seems like a Pydantic question, really.
